### DISCUSSION TITLE
Sort by formula field (static!) / Sort by relationship's field (BudibaseDB)

### DISCUSSION THREAD

**Discussion by simonedelmann** [Budibase/budibase#12014] [Help] [answered]
Hello! I am currently evaluating Budibase as a replacement for an old internal application. Looks very promising so far, but I have one blocking issue: I need to sort a table by a relationships value. 

For example: I could have a table with *TASKS*, where each of them has a relation to one *CATEGORY*. Now I want to display the tasks sorted by their category's title.

The *Data Provider* component unfortunately does not allow to sort by a relationships value. Therefore I created a **static** formula field *Category_Title* in the *TASKS* table, which does allow to display and filter the data by *Category_Title*. But sorting by *Category_Title* is NOT possible! (The field won't be displayed in the dropdown.)

If I understand correctly, static formula fields will be mirrored to the database, so sorting on database level should be possible, right?

Is there any way to archive sorting by a relationship's field? Thank you very much for your help!

(If relevant: I am using BudibaseDB and Budibase in version 2.11.15.)

**Comment by ConorWebb96** [comment]
Hey @simonedelmann,

Follow up message [here](https://discord.com/channels/733030666647765003/1161035879930925189/1161207511962107925).

**Reply by ConorWebb96** [reply, marked as answer]
Just some additional details on this one which can all be read about more extensively on the community help post linked above.

You can't sort by formula columns within the table component. There are a few ways around this; 

- For example using the views functionality and filter based on the formula column from here. 
- Additionally, you can also create an automation `Create row` that will take the newly created row, query the related row field and then update the row with the field title, this can then both be sorted on the table and anywhere else as it is now a standard text field column.

Hopefully this TLDR will help anyone who stumbles onto this Github discussion and whom don't want to go to the linked discord channel to read up on it!
